massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -360,6 +361,29 @@ impl SpeculativeLedger {
         Ok(())
     }
 
+    /// Gets a copy of a datastore keys for a given address
+    ///
+    /// # Arguments
+    /// * `addr`: address to query
+    ///
+    /// # Returns
+    /// `Some(Vec<Vec<u8>>)` for found keys, `None` if the address does not exist.
+    pub fn get_keys(&self, addr: &Address) -> Option<Vec<Vec<u8>>> {
+
+        let mut added_keys = self.added_changes.get_keys(addr);
+        let (mut final_keys, mut candidate_keys) = get_final_and_candidate_datastore_keys(
+            &self.final_state.read().ledger,
+            &self.active_history.read().0,
+            addr);
+
+        final_keys.append(&mut candidate_keys);
+        added_keys.append(&mut final_keys);
```

Comment:
It's possible to have multiple times the same key even if only one exist here ? For example if it exists in `final_state` and has been modified in `added_changes` ?

Reply:
Nop as final_keys is a BTreeSet ;)